packages/flutter_goldens_client/lib/skia_client.dart

```diff
@@ -164,34 +179,149 @@ class SkiaGoldClient extends GoldensClient {
 
     final List<String> imgtestArguments = <String>[
       'imgtest', 'add',
-      '--work-dir', _workDirectory.childDirectory('temp').path,
-      '--test-name', testName.split(path.extension(testName.toString()))[0],
+      '--work-dir', workDirectory
+        .childDirectory('temp')
+        .path,
+      '--test-name', cleanTestName(testName),
       '--png-file', goldenFile.path,
     ];
 
     await io.Process.run(
       _goldctl,
       imgtestArguments,
     );
-
-    // TODO(Piinks): Comment on PR if triage is needed, https://github.com/flutter/flutter/issues/34673
-    // So as not to turn the tree red in this initial implementation, this will
-    // return true for now.
-    // The ProcessResult that returns from line 157 contains the pass/fail
-    // result of the test & links to the dashboard and diffs.
     return true;
   }
 
+  /// Requests and sets the [_expectations] known to Flutter Gold at head.
+  Future<void> getExpectations() async {
+    await io.HttpOverrides.runWithHttpOverrides<Future<void>>(() async {
+      final Uri requestForExpectations = Uri.parse(
+        'https://flutter-gold.skia.org/json/expectations/commit/HEAD'
+      );
+      String rawResponse;
+      try {
+        final io.HttpClientRequest request = await httpClient.getUrl(requestForExpectations);
+        final io.HttpClientResponse response = await request.close();
+        rawResponse = await utf8.decodeStream(response);
+        final Map<String, dynamic> skiaJson = json.decode(rawResponse);
```

Comment:
Maybe pre-process this here into a well-known type. From the API it looks like as if we want this to be a map from golden file to List of Strings, which represent hashes.

Reply:
Fixed! :)